botocore/serialize.py

```diff
@@ -596,9 +597,64 @@ def _convert_header_value(self, shape, value):
         else:
             return value
 
+    def _finalize_serialization(self, serialized, operation_model):
+        """Perform any required post-processing steps for serialization."""
+        self._prepare_additional_traits(serialized, operation_model)
+
 
 class RestJSONSerializer(BaseRestSerializer, JSONSerializer):
 
+    def _finalize_serialization(self, serialized, operation_model):
+        super()._finalize_serialization(serialized, operation_model)
```

Comment:
This super call will do the `_prepare_additional_traits` call, which currently only calculates the `Content-MD5` checksum. There's a potential edge case where the `Content-MD5` is calculated only for the body to be modified later on this function. I don't think this is actually possible as there's no JSON services that use the `httpChecksum` trait but might be something worth noting or delaying the super call?

On that note, `_finalize_serialization` sounds like something that probably *shouldn't* be modifying the body, but rather things that would rely on the body being completely serialized (like checksums or content-length).

Reply:
Same comment as below. I'll look at getting the body modified earlier to avoid this conflict.